Flaky Test: TestBtrfs

```
--- FAIL: TestBtrfs (0.00s)
    --- FAIL: TestBtrfs/RemoveDirectoryInLowerLayer (0.14s)
        testsuite.go:96: Failed to initialize snapshotter: skip plugin
            github.com/containerd/containerd/plugin.init.ializers
            	/home/travis/gopath/src/github.com/containerd/containerd/plugin/plugin.go:38
            runtime.main
            	/home/travis/.gimme/versions/go1.12.9.linux.amd64/src/runtime/proc.go:188
            runtime.goexit
            	/home/travis/.gimme/versions/go1.12.9.linux.amd64/src/runtime/asm_amd64.s:1337
            path /tmp/snapshot-suite-Btrfs-805133001/root must be a btrfs filesystem to be used with the btrfs snapshotter
            github.com/containerd/containerd/snapshots/btrfs.NewSnapshotter
            	/home/travis/gopath/src/github.com/containerd/containerd/snapshots/btrfs/btrfs.go:80
            github.com/containerd/containerd/snapshots/btrfs.boltSnapshotter.func1
            	/home/travis/gopath/src/github.com/containerd/containerd/snapshots/btrfs/btrfs_test.go:70
            github.com/containerd/containerd/snapshots/testsuite.makeTest.func1
            	/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/testsuite.go:94
            testing.tRunner
            	/home/travis/.gimme/versions/go1.12.9.linux.amd64/src/testing/testing.go:865
            runtime.goexit
            	/home/travis/.gimme/versions/go1.12.9.linux.amd64/src/runtime/asm_amd64.s:1337
            failed to create new snapshotter
            github.com/containerd/containerd/snapshots/btrfs.boltSnapshotter.func1
            	/home/travis/gopath/src/github.com/containerd/containerd/snapshots/btrfs/btrfs_test.go:73
            github.com/containerd/containerd/snapshots/testsuite.makeTest.func1
            	/home/travis/gopath/src/github.com/containerd/containerd/snapshots/testsuite/testsuite.go:94
            testing.tRunner
            	/home/travis/.gimme/versions/go1.12.9.linux.amd64/src/testing/testing.go:865
            runtime.goexit
            	/home/travis/.gimme/versions/go1.12.9.linux.amd64/src/runtime/asm_amd64.s:1337
```

--- a/snapshots/btrfs/btrfs_test.go
+++ b/snapshots/btrfs/btrfs_test.go
@@ -26,6 +26,7 @@ import (
 	"path/filepath"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/containerd/containerd/mount"
 	"github.com/containerd/containerd/pkg/testutil"
@@ -62,11 +63,26 @@ func boltSnapshotter(t *testing.T) func(context.Context, string) (snapshots.Snap
 			loop.Close()
 			return nil, nil, errors.Wrapf(err, "failed to make btrfs filesystem (out: %q)", out)
 		}
-		if out, err := exec.Command("mount", loop.Device, root).CombinedOutput(); err != nil {
-			loop.Close()
-			return nil, nil, errors.Wrapf(err, "failed to mount device %s (out: %q)", loop.Device, out)
-		}
+		// sync after a mkfs on the loopback before trying to mount the device
+		unix.Sync()
 
+		for i := 0; i < 5; i++ {
+			if out, err := exec.Command("mount", loop.Device, root).CombinedOutput(); err != nil {
+				loop.Close()
+				return nil, nil, errors.Wrapf(err, "failed to mount device %s (out: %q)", loop.Device, out)
+			}
+			var stat unix.Statfs_t
+			if err := unix.Statfs(root, &stat); err != nil {
+				unix.Unmount(root, 0)
+				return nil, nil, errors.Wrapf(err, "unable to statfs btrfs mount %s", root)
+			}
+			if stat.Type == unix.BTRFS_SUPER_MAGIC {
+				break
+			}
+			// unmount and try again
+			unix.Unmount(root, 0)
+			time.Sleep(100 * time.Millisecond)
+		}
 		snapshotter, err := NewSnapshotter(root)
 		if err != nil {
 			loop.Close()

